### PR TITLE
add a parameter to not connect to docker during tests

### PR DESCRIPTION
**Pull request by remi-dupre**
This allows to specify `MIMIR_TEST_DOCKER__ENABLE=false` during tests to directly connect to an existing ES instance (which URL is specified through `MIMIR_TEST_ELASTICSEARCH__URL`) without attempting any operation with a docker instance.

### CODE REVIEW

File: libs/mimir/src/utils/docker.rs [hunk 1]
```diff
@@ -41,22 +43,24 @@ pub async fn initialize() -> Result<(), Error> {
...
+        if !docker.is_container_available().await? {
+            return Err(Error::Misc {
+                msg: format!(
+                    "Cannot get docker {} available",
+                    docker.docker_config.container.name
+                ),
+            });
+        }
     } else if cleanup {
         docker.cleanup().await?;
```
Comment (Liek0s) [inline libs/mimir/src/utils/docker.rs:59]: If docker is deactivated are you sure we should still call this function ?

Comment (remi-dupre) [reply libs/mimir/src/utils/docker.rs:59]: Pretty sure, the implementation of `cleanup()` bellow only connects to elasticsearch through HTTP to remove existing data. The naming is pretty bad and I'd like to change it but I deactivated my IDE tools to face my power failure :sweat:

**Comment by Liek0s** [comment]
LGTM

**Review by Liek0s — APPROVED** [review]

File: src/settings/cosmogony2mimir.rs [hunk 2]
```diff
@@ -80,11 +80,20 @@ pub enum Command {
 impl Settings {
     // Read the configuration from <config-dir>/cosmogony2mimir and <config-dir>/elasticsearch
     pub fn new(opts: &Opts) -> Result<Self, Error> {
+        let prefix = {
+            if opts.run_mode.as_deref() == Some("testing") {
+                "MIMIR_TEST"
+            } else {
+                println!("helo");
```
Comment (sdrll) [inline src/settings/cosmogony2mimir.rs]: hello :)

File: src/settings/ntfs2mimir.rs [hunk 3]
```diff
@@ -89,11 +89,20 @@ pub enum Command {
 impl Settings {
     // Read the configuration from <config-dir>/ntfs2mimir and <config-dir>/elasticsearch
     pub fn new(opts: &Opts) -> Result<Self, Error> {
+        let prefix = {
+            if opts.run_mode.as_deref() == Some("testing") {
+                "MIMIR_TEST"
+            } else {
+                println!("helo");
```
Comment (sdrll) [inline src/settings/ntfs2mimir.rs]: Hello x2

File: src/settings/openaddresses2mimir.rs [hunk 4]
```diff
@@ -102,11 +102,20 @@ pub enum Command {
 impl Settings {
     // Read the configuration from <config-dir>/openaddresses2mimir and <config-dir>/elasticsearch
     pub fn new(opts: &Opts) -> Result<Self, Error> {
+        let prefix = {
+            if opts.run_mode.as_deref() == Some("testing") {
+                "MIMIR_TEST"
+            } else {
+                println!("helo");
```
Comment (sdrll) [inline src/settings/openaddresses2mimir.rs]: hello :)

File: src/settings/osm2mimir.rs [hunk 5]
```diff
@@ -115,11 +115,20 @@ pub enum Command {
 impl Settings {
     // Read the configuration from <config-dir>/osm2mimir and <config-dir>/elasticsearch
     pub fn new(opts: &Opts) -> Result<Self, Error> {
+        let prefix = {
+            if opts.run_mode.as_deref() == Some("testing") {
+                "MIMIR_TEST"
+            } else {
+                println!("helo");
```
Comment (sdrll) [inline src/settings/osm2mimir.rs]: Hello !

File: src/settings/poi2mimir.rs [hunk 6]
```diff
@@ -96,11 +96,20 @@ pub enum Command {
 impl Settings {
     // Read the configuration from <config-dir>/osm2mimir and <config-dir>/elasticsearch
     pub fn new(opts: &Opts) -> Result<Self, Error> {
+        let prefix = {
+            if opts.run_mode.as_deref() == Some("testing") {
+                "MIMIR_TEST"
+            } else {
+                println!("helo");
```
Comment (sdrll) [inline src/settings/poi2mimir.rs]: hello :)

**Pull request merged**